bokehjs/src/lib/core/property_mixins.ts

```diff
@@ -18,6 +18,15 @@ export type Fill = {
   fill_alpha: p.Property<number>
 }
 
+export type Hatch = {
+  hatch_color: p.Property<Color | null>
+  hatch_alpha: p.Property<number>
+  hatch_scale: p.Property<number>
+  hatch_pattern: p.Property<string>
+  hatch_weight: p.Property<number>
+  hatch_extra: p.Any
```

Comment:
Property implementations should be left for `define()`. In types like this one, you should preferably (there are exceptions related to specs) use abstract property types, so `p.Property<unknown>`. However, given that `Dict(String, Instance("bokeh.models.callbacks.CustomJS"))`  can be expressed in the type system, the complete type is `p.Property<{[key: string]: CustomJS}>`. Then you can use `p.Any` in `define()`, because we don't have a specific implementation for this kind of property.